### PR TITLE
bug: use custom ante handler to reject redundant transactions in simapp

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Adds usage of the custom ante handler that rejects redundant transactions (when in different block, at the moment) in simapp.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: testing/simapp/ante_handler.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// HandlerOptions extend the SDK's AnteHandler options by requiring the IBC
+// channel keeper.
+type HandlerOptions struct {
+	ante.HandlerOptions
+
+	IBCChannelkeeper channelkeeper.Keeper
+}
+
+func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
```
Comment (damiannolan) [inline testing/simapp/ante_handler.go:20]: Maybe add a godoc for `NewAnteHandler` :)

**Review by damiannolan — APPROVED** [review]
🚀

**Review by colin-axner — APPROVED** [review]
LGTM, could you add a changelog entry?

**Review by seantking — APPROVED** [review]
nice !

**Review by AdityaSripal — APPROVED** [review]

**Comment by seantking** [comment]
Should we merge this?

**Pull request merged**